lib/BasicAnalyses/PromoteGlobalToLocalVars.cpp

```diff
@@ -0,0 +1,50 @@
+/// \file PromoteGlobalToLocalVars.cpp
+/// \brief Promote CSVs in form of global variables to local variables.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/IRBuilder.h"
+
+#include "revng/BasicAnalyses/PromoteGlobalToLocalVars.h"
+#include "revng/Support/IRHelpers.h"
+
+llvm::PreservedAnalyses
+PromoteGlobalToLocalPass::run(llvm::Function &F,
+                              llvm::FunctionAnalysisManager &FAM) {
```

Comment:
It'd be nice to be able to run this pass from `opt`.
Can you add a check that ignores all non-isolated functions?
You can use `FunctionTags`.

Same for `RemoveHelperCalls`.

Reply:
I believe `PromoteGlobalToLocalPass` should not be meant to be used anywhere but in SA2. There is `PromoteCSVs` for isolated functions. Also, I'am in that phase of the pipeline in which the isolated functions supposedly do not exist yet, aren't I?

Reply:
Ah right.
If it's supposed to be used in SA2 only, then move it in `lib/StackAnalysis`.

Reply:
Then I should move `SegregateDirectStackAccessesPass` in `lib/StackAnalysis` as well, shouldn't I?

Reply:
Makes sense.